### PR TITLE
feat(donation): Remove pages related to payment and donations

### PR DESCRIPTION
**Pull request by ERosendo**
This PR introduces the following changes:

1. Removes the `/pay/` page.

2. Remove the donation page and update all references to it with the new donation URL: https://donate.free.law/forms/supportflp

3. Removes the page to register check donations.

4. Removes the /badges/sign-up/ page

5. Replaces the old contribution list in the "Donations" tab with a link to the new Neon dashboard login. Here's a screenshot of the new tab: 

image

### CODE REVIEW

File: cl/donate/templates/donate.html
Comment (mlissner) [inline cl/donate/templates/donate.html:1]: Can't this whole page be removed?

File: cl/donate/templates/donate_complete.html
Comment (mlissner) [inline cl/donate/templates/donate_complete.html:1]: Can't this whole page go away?

File: cl/donate/tests.py
Comment (mlissner) [inline cl/donate/tests.py:1]: I think this whole file can go.

File: cl/donate/urls.py
Comment (mlissner) [inline cl/donate/urls.py:1]: I think everything in this file can go except for the callbacks, which we need until we migrate our monthly payments to Neon.

This path and its logic can get moved to the user app:

```
path("profile/donations/", view_donations, name="profile_donations"),
```

File: cl/donate/utils.py
Comment (mlissner) [inline cl/donate/utils.py:1]: Everything goes from here, except for code related to monthly donations.

File: cl/donate/views.py
Comment (mlissner) [inline cl/donate/views.py:1]: I think this whole file can go.

File: cl/users/templates/profile/donations.html
Comment (mlissner) [inline cl/users/templates/profile/donations.html:28]: This is really good. How about:

```suggestion
        Your generosity deserves transparency. We use Neon Pay to handle donations and you can create an account in their system to access your detailed donation list, including one-time and monthly contributions. See how you're making a difference.
```

File: cl/users/templates/profile/donations.html
Comment (mlissner) [inline cl/users/templates/profile/donations.html:34]:
```suggestion
        If you have any questions, please get in touch <a href="mailto:donate@free.law">donate@free.law</a>. We'd be happy to hear from you.
```

**Review by mlissner — CHANGES_REQUESTED** [review]
This is good, but we need to be even more aggressive about excising the evils of payments from CourtListener. Everything must go with the exception of code that completes monthly payments. 

The migration will be like this:

1. Land webhooks and membership stuff so we know when people sign up on Neon.

1. Stop allowing donations in CourtListener and redirect all payments to Neon.

3. Work with Neon staff to migrate all monthly payment info.

4. Remove Stripe and PayPal from CL completely.

The one thing that excited me more than removing code is removing payment processing code.

File: cl/donate/templates/donate.html
Comment (ERosendo) [reply cl/donate/templates/donate.html:1]: Yes, you're right. I'm removing this template

File: cl/donate/templates/donate_complete.html
Comment (ERosendo) [reply cl/donate/templates/donate_complete.html:1]: Yup

File: cl/donate/urls.py
Comment (ERosendo) [reply cl/donate/urls.py:1]: > This path and its logic can get moved to the user app:

got it!

File: cl/users/templates/profile/donations.html
Comment (ERosendo) [reply cl/users/templates/profile/donations.html:28]: Thanks for the suggestion. I'll update the template.

File: cl/donate/tests.py
Comment (ERosendo) [reply cl/donate/tests.py:1]: I kept the test related to Stripe and monthly donations in this file, is it ok to remove all of them?

Comment (mlissner) [reply cl/donate/tests.py:1]: Sure, might as well keep that, I guess. :) Doesn't really matter since we won't be changing or working on that code, but best practice is surely to keep it until its functionality is excised.

File: cl/donate/templates/donate_complete.html
Comment (ERosendo) [reply cl/donate/templates/donate_complete.html:1]: I thought we could remove this template, but we also use it in other places such as `process_paypal_callback` and `donate_paypal_cancel` views.

Comment (mlissner) [reply cl/donate/templates/donate_complete.html:1]: But we're removing paypal payments too, right?

**Comment by mlissner** [comment]
- [ ] One other thought here: We have a lot of links all over the Internet that go to `/donate/`, so we should make sure that redirects to free.law/donate/, rather than *entirely* removing it.

**Comment by ERosendo** [comment]
@mlissner I've applied all your suggestions and this PR is ready for another review. Here's a summary of the changes:

- Removes the following templates: `check_donation.html`, `badge_signup.html`, `badge_signup_complete.html`, `cc_payment.py`, `donate.html`, `payment_base.html` and `payment_complete.py`

- Deactivates PayPal integration by removing callback and helper functions (paypal.py file). 

- Removes the logic to turn off monthly donations and tweaks the `donation_thanks_recurring` email template. 

- Move the profile_donations URL to the user's app

- Remove email templates related to donations, payments, and badges.

- Updates the `send_thank_you_email` method to just handle monthly donations emails. 

- Adds redirection to `free.law/donate/` for the `/donate` view

- Adds links to the membership form in multiple HTML templates. 

### Questions 

- I tried to remove the `views.py` file from the `donate` app but I noticed that the `payment_complete` view is currently being used in the `charge_monthly_donors` command. Are we using this command? 

Let me know what you think.

File: cl/donate/templates/donate_complete.html
Comment (mlissner) [inline cl/donate/templates/donate_complete.html:1]: We get to remove this, I think.

File: cl/donate/templates/includes/donate_sidebar.html
Comment (mlissner) [inline cl/donate/templates/includes/donate_sidebar.html:1]: I think we should tweak this to say membership information, since we won't have donation information anymore. 

So for members:

> ## Thank you!
> 
> You're a {tier-level} level supporter of the Free Law Project. Thank you for your support! 

And for non-members it can be the same, but instead of the "We rely..." sentence, it can say:

> Members help support our work and get special access to features.
> 
> Please become a member today.
> 
> [ Join Free.law Now ]

File: cl/donate/urls.py
Comment (mlissner) [inline cl/donate/urls.py:9]: I usually put all our one-off legacy redirects in cl/urls.py. If we do that here, we can remove this file once the stripe callback is removed.

File: cl/donate/utils.py
Comment (mlissner) [inline cl/donate/utils.py:1]: We can (and should) remove `get_donation_totals_by_email`. It's only used when people use the contact link, and Brian wants it removed from there anyway (it's a bit tacky). You'll fix https://github.com/freelawproject/courtlistener/issues/2898 when you do.

File: cl/lib/context_processors.py
Comment (mlissner) [inline cl/lib/context_processors.py:38]: Time for a stats update by 100×.

```suggestion
    'CourtListener serves more than 200 million requests per year, but has a lean staff of only a handful of developers. Please <a href="%s?referrer=tip">donate</a> to support our work making the judiciary more equitable. <i class="fa fa-heart red"></i>'
```

File: cl/opinion_page/templates/includes/docket_alerts_modal.html
Comment (mlissner) [inline cl/opinion_page/templates/includes/docket_alerts_modal.html:1]: This file needs to be updated to say "Member" throughout. Can you take a stab at it, please?

File: cl/search/templates/advanced.html
Comment (mlissner) [inline cl/search/templates/advanced.html:604]: Can you make this say "please donate or become a member so...", and then add a second button next to the first one for [Join Free.law] ?

File: cl/search/templates/advanced.html
Comment (mlissner) [inline cl/search/templates/advanced.html:662]: Same here. "Please donate or become a member," and then two buttons?

File: cl/search/templates/advanced.html
Comment (mlissner) [inline cl/search/templates/advanced.html:737]: And here too, please.

File: cl/search/templates/includes/alert_modal.html
Comment (mlissner) [inline cl/search/templates/includes/alert_modal.html:71]: Let's tweak the button with:

```suggestion
                   target="_blank">Join Free.law</a>
```

And the paragraph above should be tweaked to say:

```
<a href="xxx">Real Time alerts</a> are only supported for members of <a href="https://free.law/">Free Law Project</a>.
```

And finally, this line:

>               <p>Select another option or donate to create this alert.</p>

Can be:

>               <p>Please select another rate or become a member to create this alert.</p>

File: cl/simple_pages/templates/contribute.html
Comment (mlissner) [inline cl/simple_pages/templates/contribute.html:29]: Tweaks, please:

```
  <p>If you are looking to support our work, the first thing you can do to help is donate to Free Law Project or join us as a member. This allows us to hire more full time staff. It can be uncomfortable to say so directly, but nothing makes as big a difference as financial support. Please consider donating or becoming a member if you can.
  </p>
```

And then we'll need the double buttons again as elsewhere.

File: cl/simple_pages/templates/help/alert_help.html
Comment (mlissner) [inline cl/simple_pages/templates/help/alert_help.html:45]: Let's push memberships here too:

```
 <p>We hope that if you find our alerts useful, you will become a member of <a href="https://free.law">Free Law Project</a>, the non-profit supporting CourtListener and RECAP. It is the best thing you can do to support ongoing feature development and maintenance.
    </p>
```

File: cl/simple_pages/templates/help/alert_help.html
Comment (mlissner) [inline cl/simple_pages/templates/help/alert_help.html:46]: And a tweak here, of course.

File: cl/simple_pages/templates/help/alert_help.html
Comment (mlissner) [inline cl/simple_pages/templates/help/alert_help.html:80]:
```suggestion
      <a href="https://donate.free.law/forms/membership"
         class="btn btn-default"><i class="fa fa-heart-o"></i> Become a Free.law Member</a>
    </p>
```

File: cl/simple_pages/templates/help/alert_help.html
Comment (mlissner) [inline cl/simple_pages/templates/help/alert_help.html:241]: This pricing is no longer correct. Now they have to be monthly or legacy donors. So:

```suggestion
    <p>Real Time alerts are only available for supporters of Free Law Project. If you see a message like the one below, you will <a href="https://donate.free.law/forms/membership">need to become a member</a> to enable Real Time alerts:
```

File: cl/simple_pages/templates/help/alert_help.html
Comment (mlissner) [inline cl/simple_pages/templates/help/alert_help.html:244]: This image will need to be updated.

File: cl/simple_pages/templates/help/coverage.html
Comment (mlissner) [inline cl/simple_pages/templates/help/coverage.html:143]: I'd do a double button here. One for donate and one for [Join Free.law]

File: cl/donate/views.py
Comment (ERosendo) [reply cl/donate/views.py:1]: You were right!

File: cl/simple_pages/templates/help/coverage_fds.html
Comment (mlissner) [inline cl/simple_pages/templates/help/coverage_fds.html:217]: Another double button here, please.

File: cl/simple_pages/templates/help/recap_email_help.html
Comment (mlissner) [inline cl/simple_pages/templates/help/recap_email_help.html:69]: I'm sorry, I think both of these should be `supportflp` links. :) 

I know, I'm at 34 comments and it's too many!

File: cl/simple_pages/templates/help/tags_help.html
Comment (mlissner) [inline cl/simple_pages/templates/help/tags_help.html:40]: Let's encourage memberships here though, please.

File: cl/simple_pages/templates/podcasts.html
Comment (mlissner) [inline cl/simple_pages/templates/podcasts.html:25]: Double button, please.

File: cl/users/templates/profile/donations.html
Comment (mlissner) [inline cl/users/templates/profile/donations.html:1]: Somewhere on this page we need to tell people if they are members and say thank you. It can be pretty simple. Notionally:

> Thank you for being a member of Free Law Project! 

Instead of:

> Thank You for Your Support!

? 

Want to take a stab at something like that?

**Review by mlissner — CHANGES_REQUESTED** [review]
Well....I found a lot to comment about, I'm sorry, but I think it's basically all tiny stuff to try to get the switch to memberships right. 

Basically looks good though, thank you!

Sorry, editing to say one other thing. We have more payment-related stuff in the `UserProfile` model:

I think this can get removed:
```
    @property
    def total_donated_last_year(self) -> Decimal:
        one_year_ago = now() - timedelta(days=365)
        total = (
            self.user.donations.filter(date_created__gte=one_year_ago)
            .exclude(status__in=donation_exclusion_codes)
            .aggregate(Sum("amount"))["amount__sum"]
        )
        if total is None:
            total = Decimal(0.0)
        return total

    @property
    def total_donated(self) -> Decimal:
        total = self.user.donations.exclude(
            status__in=donation_exclusion_codes
        ).aggregate(Sum("amount"))["amount__sum"]
        if total is None:
            total = Decimal(0.0)
        return total

    @property
    def is_monthly_donor(self) -> bool:
        """Does the profile have any monthly donations set up and running?

        :return bool: True if so, False if not.
        """
        return bool(self.user.monthly_donations.filter(enabled=True).count())
```

I think this needs to be tweaked to look at membership info:
```
    @property
    def can_make_another_alert(self) -> bool:
        """Can the user make another alert?

        The answer is yes, if any of the following is true:
         - They get unlimited ones
         - They are a monthly donor # <---- NO LONGER TRUE
         - They are under the threshold
         - Their email domain is unlimited

        return: True if they can make another alert; else False.
        """
        if any(
            [
                # Place performant checks first
                self.unlimited_docket_alerts,
                self.email_grants_unlimited_docket_alerts,
                self.is_monthly_donor,
                self.user.docket_alerts.subscriptions().count()
                < settings.MAX_FREE_DOCKET_ALERTS,
            ]
        ):
            return True
        return False
```

File: cl/donate/urls.py
Comment (ERosendo) [reply cl/donate/urls.py:9]: > I usually put all our one-off legacy redirects in cl/urls.py.

understood!

File: cl/donate/utils.py
Comment (ERosendo) [reply cl/donate/utils.py:1]: got it!

File: cl/opinion_page/templates/includes/docket_alerts_modal.html
Comment (ERosendo) [reply cl/opinion_page/templates/includes/docket_alerts_modal.html:1]: sure!

File: cl/search/templates/advanced.html
Comment (ERosendo) [reply cl/search/templates/advanced.html:604]: Yes of course!

File: cl/search/templates/advanced.html
Comment (ERosendo) [reply cl/search/templates/advanced.html:662]: Sure!

**Comment by ERosendo** [comment]
@mlissner I've applied your suggestion and the PR is ready for another review :). 

This PR depends on #3468 now, we need the membership model introduced in #3468 to remove and replace the legacy donation code.

File: cl/users/templates/profile/alerts.html
Comment (mlissner) [inline cl/users/templates/profile/alerts.html:91]: This needs updating: "Please become a member to enable Real Time alerts."

File: cl/users/templates/profile/alerts.html
Comment (mlissner) [inline cl/users/templates/profile/alerts.html:89]: Doesn't this logic need flipping?

File: cl/users/templates/profile/donations.html
Comment (mlissner) [inline cl/users/templates/profile/donations.html:29]: "These and other features" is plural, so:

```suggestion
        Your ongoing membership fuels our passion to keep developing these and other features that make the judiciary more equitable.
```

**Review by mlissner — CHANGES_REQUESTED** [review]
Super close, thank you.

File: cl/users/templates/profile/alerts.html
Comment (ERosendo) [reply cl/users/templates/profile/alerts.html:89]: You're right!

File: cl/users/templates/profile/donations.html
Comment (ERosendo) [reply cl/users/templates/profile/donations.html:29]: oh sorry. Thank you for bringing it to my attention.

**Comment by ERosendo** [comment]
@mlissner I used your feedback to tweak the alerts.html template and fix the typo in the donations tab, I've also removed the donate help page and users will now be redirected to the help section at free.law/donate/help/. This page will be available once we merge freelawproject/free.law#161.

**Review by mlissner — APPROVED** [review]
LGTM. We can merge this once we've got it properly sequenced behind the webhook PR.

Thank you @ERosendo for deleting so much code!

File: cl/users/templates/profile/donations.html
Comment (mlissner) [inline cl/users/templates/profile/donations.html:1]: Working on some of this stuff. Can we update the pill for this to be "Your Support" instead of "Donations", since this has both membership info as well as donations? 

And can we update the HTML `<title>` tag too? 

(The URL is still inaccurate, but it's close enough.)

Comment (ERosendo) [reply cl/users/templates/profile/donations.html:1]: > Can we update the pill for this to be "Your Support" instead of "Donations", since this has both membership info as well as donations?

Sure!

**Pull request merged**
